src/main/scala/is/hail/distributedmatrix/RowMatrix.scala

```diff
@@ -105,4 +161,96 @@ class RowMatrix(val hc: HailContext,
       }.filter(_.nonEmpty)
     }.writeTable(path, hc.tmpDir, header, exportType)
   }
-}
\ No newline at end of file
+}
+
+// [`start`, `end`) is the row range of partition
+case class ReadBlocksAsRowsRDDPartition(index: Int, start: Long, end: Long) extends Partition
+
+class ReadBlocksAsRowsRDD(path: String,
+  sc: SparkContext,
+  partitionCounts: Array[Long],
+  gp: GridPartitioner) extends RDD[(Long, Array[Double])](sc, Nil) {
+  
+  private val partitionStarts = partitionCounts.scanLeft(0L)(_ + _)
+  
+  if (partitionStarts.last != gp.nRows)
+    fatal(s"Error reading BlockMatrix as RowMatrix: expected ${partitionStarts.last} rows in RowMatrix, but found ${gp.nRows} rows in BlockMatrix")
+
+  if (gp.nCols > Int.MaxValue)
+      fatal(s"Cannot read BlockMatrix with ${gp.nCols} > Int.MaxValue columns as a RowMatrix")
+  
+  private val nCols = gp.nCols.toInt
+  private val nBlockCols = gp.nBlockCols
+  private val blockSize = gp.blockSize
+
+  private val d = digitsNeeded(gp.numPartitions)
+  private val sHadoopBc = sc.broadcast(new SerializableHadoopConfiguration(sc.hadoopConfiguration))
+
+  protected def getPartitions: Array[Partition] = Array.tabulate(partitionStarts.length - 1)(pi => 
+    ReadBlocksAsRowsRDDPartition(pi, partitionStarts(pi), partitionStarts(pi + 1)))
```

Comment:
I'd just call it `i` or `index`

Reply:
You're BlockMatrix code has...evolved. Cotton and I settled on pi for partition index.  In this class, I consistently use i for the row index.